packages/shared/lib/contexts/onboarding/actions/initialiseOnboardingProfile.ts

```diff
@@ -0,0 +1,21 @@
+import { get } from 'svelte/store'
+
+import { AppStage, appStage } from '@core/app'
+import { profileManager } from '@core/profile-manager'
+
+import { buildOnboardingProfile } from '../helpers'
+import { onboardingProfile } from '../stores'
+
+/**
+ * Builds a new onboarding profile and sets the Svelte store accordingly.
+ */
+export function initialiseOnboardingProfile(): Promise<void> {
+    if (get(profileManager)) {
+        console.error('Profile is already created')
+        return
+    }
+
+    const isDeveloperProfile = get(appStage) !== AppStage.PROD
```

Comment:
Why are we hardcoding this here? It should be passed in depending where the user clicks the button. i.e. through the electron menu or through profile selection page.